src/ucp/wireup/wireup_cm.c

```diff
@@ -439,51 +490,38 @@ static unsigned ucp_cm_client_uct_connect_progress(void *arg)
     UCS_STATIC_ASSERT(ucs_static_array_size(ep_init_flags_prio) > 0);
 
     ucp_wireup_eps_pending_extract(ep, &tmp_pending_queue);
+
+    /* Cleanup the previously created UCP EP. The one that was created on
+     * the previous call to this client's resolve_cb */
+     ucp_wireup_cm_ep_cleanup(ep);
+
     for (i = 0; i < ucs_static_array_size(ep_init_flags_prio); ++i) {
         ep_init_flags = ep_init_flags_prio[i];
 
-        /* Cleanup the previously created UCP EP. The one that was created on
-         * the previous call to this client's resolve_cb */
-        ucp_wireup_cm_ep_cleanup(ep);
-
         status = ucp_cm_ep_client_initial_config_get(ep, ep_init_flags,
                                         &cm_wireup_ep->cm_resolve_tl_bitmap,
                                         &key);
         if (status != UCS_OK) {
             goto try_fallback;
         }
 
-        ucp_ep_realloc_lanes(ep, key.num_lanes);
-
         status = ucp_worker_get_ep_config(worker, &key, ep_init_flags,
                                           &ep->cfg_index);
         if (status != UCS_OK) {
             goto err;
         }
 
-        ep->am_lane = key.am_lane;
-
-        status = ucp_cm_ep_init_lanes(ep, &tl_bitmap);
-        if (status != UCS_OK) {
-            goto err;
-        }
-
-        if (i < (ucs_static_array_size(ep_init_flags_prio) - 1)) {
-            /* Can use another variant of endpoint's initialization flags for
-             * fallback */
-            fallback_log_level = UCS_LOG_LEVEL_DEBUG;
-        } else if (ucp_cm_client_get_next_cm_idx(ep) != UCP_NULL_RESOURCE) {
+        fallback_log_level = cm_pack_log_level[i];
+        if ((i == ucs_static_array_size(ep_init_flags_prio) - 1) &&
+            (ucp_cm_client_get_next_cm_idx(ep) != UCP_NULL_RESOURCE)) {
```

Comment:
maybe test `(ucp_cm_client_get_next_cm_idx(ep) == UCP_NULL_RESOURCE))` and set `UCS_LOG_LEVEL_ERROR` only for last case? and debeug,debug,diag in the prio array

Reply:
@evgeny-leksikov Agree. Done.

Reply:
passed all CI test cases